### PR TITLE
feat: add create database from share SQL

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

feat: add create database from share SQL

Fixes [#issue](https://github.com/datafuselabs/databend/issues/7269)

### CODE REVIEW

**Review by drmingdrmer — APPROVED** [review]

**Comment by drmingdrmer** [comment]
`main` branch is force pushed(**again** `:(`, to cleanup big object in history), update your working branch please: [To everybody: main branch is force pushed. #7263](https://github.com/datafuselabs/databend/issues/7263)

**Comment by lichuang** [comment]
@Xuanwo

File: src/query/ast/src/parser/statement.rs [hunk 1]
```diff
@@ -169,16 +176,39 @@ pub fn statement(i: Input) -> IResult<StatementMsg> {
...
-                if_not_exists: opt_if_not_exists.is_some(),
-                catalog,
-                database,
-                engine,
-                options: vec![],
-            })
+            CREATE ~ ( DATABASE | SCHEMA ) ~ ( IF ~ NOT ~ EXISTS )? ~ #peroid_separated_idents_1_to_2 ~ #create_database_option?
+        },
+        |(_, _, opt_if_not_exists, (catalog, database), create_database_option)| {
+            match create_database_option {
```
Comment (Xuanwo) [inline src/query/ast/src/parser/statement.rs:182]: Is it possible to specify the engine and from share at the same time?

File: src/query/service/src/sql/planner/binder/insert.rs [hunk 2]
```diff
@@ -462,7 +466,7 @@ async fn exprs_to_datavalue<'a>(
         );
         let (mut scalar, data_type) = scalar_binder.bind(expr).await?;
         let field_data_type = schema.field(i).data_type();
-        if data_type.ne(field_data_type) {
+        if !data_type.eq(field_data_type) {
```
Comment (Xuanwo) [inline src/query/service/src/sql/planner/binder/insert.rs:469]: How about using `!=` directly?

**Comment by Xuanwo** [comment]
> `main` branch is force pushed(**again** `:(`, to cleanup big object in history), update your working branch please: [To everybody: main branch is force pushed. #7263](https://github.com/datafuselabs/databend/issues/7263)

This PR contains unrelated commits. Please fix it.

**Pull request closed without merging**